docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,184 @@
+
+- If `scheduled` is `"scheduled"`, then this TripKey's `tripId` will equal the `trips_updated` TripKey's `tripId`.
+- If `scheduled` is `"added"`, then this TripKey's `tripId` will equal the `trips_updated` TripKey's `glidesId`.
+- (In both cases, the `serviceDate` must also match in order to know the two TripKeys refer to the same trip.)
+
+The difference with the TripKey in `trips_updated` is because `trips_updated` has to give enough information about a trip that a consumer can understand it even without recognizing the trip ID, while `vehicle_trip_assignments` assumes that you can look up information in `trips_updated` or in `GTFS`, so doesn't give as much information as possible about the trip. (Though consumers should keep in mind that order of events between two event streams is not guaranteed, and a trip that's referenced in this feed may not have appeared in `trips_updated`.)
+
+For scheduled trips, the `tripId` is intended to line up to the trip IDs in GTFS most of the time, because they're derived from the same source (HASTUS's `trp_int_number`). However, Glides may be using a slightly different schedule than any consumer (due to disruptions, for example), so matching GTFS is not guaranteed. If a consumer sees a `tripId` in this feed that also appears in GTFS on a trip on the same service date, then it refers to the same trip.
+
+For added trips, Glides makes up a new ID, which is kept consistent for any subsequent updates to the same added trip. Glides tends to prefix trip IDs for added trips with the string `"ADDED-"` but consumers MUST NOT assume any particular format, or assume that they can tell an ID for an added trip apart from a scheduled trip ID by the string alone. That's what the `scheduled` field is for.
+
+The added/scheduled distinction is according to the schedule in Glides. If Glides has a different schedule than RTR or GTFS, then a trip that Glides thinks is scheduled may not match up with scheduled trip in GTFS, and RTR (or other consumers) may decide that a trip that was added in Glides lines up close enough to correspond to a scheduled trip in GTFS.
+
+Revenue is included because this is the most important information that Glides can provide to riders about a train, and its presence will make it less likely for a consumer of this feed to incorrectly interperet a nonrevenue train as being in service. Especially since nonrevenue trips are not in GTFS and their IDs are likely to not be recognized by a consumer.
+
+Consumers SHOULD use the use the entire object when comparing two `TripKey`s to see if they're equal, not just the `tripId` field. Otherwise, updates could be applied to the wrong trip, for example, updating the vehicle assignment for yesterday's trip instead of today's.
```

Comment:
I think this is related to [my question about car numbers](https://github.com/mbta/schemas/pull/9#discussion_r1843922483) as well - are we saying that there may be events that differ only in fields like revenue or car numbers and should be treated as duplicates, or are we saying that the consumer should apply updates to those respective fields even if the trip key / vehicle ID combo stays the same?

Reply:
Ooh, good question, and I guess this problem is one of the tradeoffs in including the revenue field even though it's not required. I'm not sure.

For scheduled trips, revenue is part of the schedule and shouldn't change (and if it does, that's specified in trips_updated), so I'm fine treating it as part of the ID.

For added trips, it's more likely for the revenue status to change. In that case it's more like a payload than the ID. But this is a bad format for publishing updates to trip data. If revenue was part of the ID, then if it was ever changed, then you'd have two different added trips with the same `{serviceDate, tripId, scheduled: "added"}`, which could cause problems if you weren't consistent about always including revenue in the ID.

But if added trips didn't have the revenue field, then if a consumer hadn't seen that trip come through trips_updated, then they'd have no idea if the vehicle was in revenue service or not.

I think including revenue in the ID so the whole TripKey is the ID is the most straightforward answer, so I'm leaning that way, but I'm not certain.

Reply:
After talking with Eddie and Walton, I've removed the revenue field in 22245e1.

RTR should get the revenue status (and other trip info) from the trips_updated feed. RTR should log if they fail to find data there.

If it becomes a problem, we could reinstate the revenue field, though it should probably be a top-level field within `data` rather than part of the `TripKey`. (In that case, it'd be more like a field of the vehicle "is the vehicle accepting passengers" rather than a field on the trip.)

Reply:
If we need to ensure that the trip exists in the `trips_updated` feed before sending any `vehicle_trip_assignment`s, I had an idea for one way we could do that: https://github.com/mbta/schemas/pull/10

Reply:
Also, for the record, this is only really an issue for _scheduled_ non-revenue trips, because those are the ones that RTR wouldn't have via GTFS or via TripUpdates (barring any edits made to them). However, the only scheduled non-rev trips we really have at the moment that go on what's normally revenue trackage and therefore need to be suppressed are Inner Belt pullouts / pullbacks, and there is already a mechanism for suppressing those. Also, in the long run, RTR can learn about scheduled non-revenue trips via TODs.